src/audio/google/google_rtc_audio_processing.c

```diff
@@ -94,19 +93,16 @@ static void google_rtc_audio_processing_params(struct processing_module *mod)
 
 	ipc4_base_module_cfg_to_stream_params(&mod->priv.cfg.base_cfg, params);
 	component_set_nearest_period_frames(dev, params->rate);
-
-	list_for_item(source_list, &dev->bsource_list) {
-		sourceb = container_of(source_list, struct comp_buffer, sink_list);
-		if (IPC4_SINK_QUEUE_ID(source->id) == SOF_AEC_FEEDBACK_QUEUE_ID)
-			ipc4_update_buffer_format(sourceb, &cd->config.reference_fmt);
-		else
-			ipc4_update_buffer_format(sourceb, &mod->priv.cfg.base_cfg.audio_fmt);
-	}
-
-	sinkb = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-	ipc4_update_buffer_format(sinkb, &mod->priv.cfg.base_cfg.audio_fmt);
+	/* TODO - it does not work, to be checked before merging!!
+	 * ipc4_update_source_format(ref, &cd->config.reference_fmt);
+	 */
+	source_set_channels(ref,
+				CONFIG_COMP_GOOGLE_RTC_AUDIO_PROCESSING_NUM_AEC_REFERENCE_CHANNELS);
+	/* ipc4_update_source_format(mic, &mod->priv.cfg.base_cfg.audio_fmt); */
+	source_set_channels(mic, CONFIG_COMP_GOOGLE_RTC_AUDIO_PROCESSING_NUM_CHANNELS);
+	/* ipc4_update_sink_format(out, &mod->priv.cfg.base_cfg.audio_fmt); */
```

Comment:
Why aren't we using the config from the stream as number of channels ?
We could use the stream parameter to configure the number of channels on the AEC.

Reply:
this PR is not yet finished
Currently - for unknown reasons - the number of channels at inputs are set incorrectly and I has to be overwritten. Root cause is under investigation.